ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_activations.py

```diff
@@ -241,34 +307,458 @@ def test_logsigmoid(
+        rtol_=1e-1,
+        atol_=1e-1,
+        x=x,
+        weight=weight,
+    )
+
+
+# celu
+@handle_test(
+    fn_tree="functional.ivy.experimental.celu",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float", full=False, key="leaky_relu"),
+        large_abs_safety_factor=16,
+        small_abs_safety_factor=16,
+        safety_factor_scale="log",
+    ),
+    alpha=st.floats(min_value=1e-4, max_value=1.0),
+)
+def test_celu(
+    *,
+    dtype_and_x,
+    alpha,
+    test_flags,
+    backend_fw,
+    fn_name,
+    on_device,
+    ground_truth_backend,
+):
+    dtype, x = dtype_and_x
+    helpers.test_function(
+        ground_truth_backend=ground_truth_backend,
+        input_dtypes=dtype,
+        fw=backend_fw,
+        test_flags=test_flags,
+        fn_name=fn_name,
+        on_device=on_device,
+        rtol_=1e-1,
+        atol_=1e-1,
+        x=x[0],
+        alpha=alpha,
+    )
+
+
+# glu
+@handle_test(
+    fn_tree="functional.ivy.experimental.glu",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("float"),
+        valid_axis=True,
+        max_axes_size=1,
+        force_int_axis=True,
+    ).filter(lambda x: x[2] is not None and x[1][0].shape[x[2]] % 2 == 0),
+)
+def test_glu(
+    *,
+    dtype_x_axis,
+    test_flags,
+    backend_fw,
+    fn_name,
+    on_device,
+    ground_truth_backend,
+):
+    input_dtypes, x, axis = dtype_x_axis
+    axis = axis if isinstance(axis, int) else -1
+    helpers.test_function(
+        ground_truth_backend=ground_truth_backend,
+        input_dtypes=input_dtypes,
+        fw=backend_fw,
+        test_flags=test_flags,
+        fn_name=fn_name,
+        on_device=on_device,
+        x=x[0],
+        axis=axis,
+        rtol_=1e-1,
+        atol_=1e-1,
+    )
+
+
+# group_norm
+@handle_test(
+    fn_tree="functional.ivy.experimental.group_norm",
+    dtype_x_and_axis=_generate_data_layer_norm(
+        available_dtypes=helpers.get_dtypes("float"),
+        min_num_dims=2,
+        max_num_dims=3,
+        min_dim_size=2,
+        max_dim_size=4,
+        group=True,
+    ),
+    eps=st.floats(min_value=0.01, max_value=0.1),
+    test_with_out=st.just(False),
+)
+def test_group_norm(
+    *,
+    dtype_x_and_axis,
+    eps,
+    test_flags,
+    backend_fw,
+    fn_name,
+    on_device,
+    ground_truth_backend,
+):
+    dtype, x, weight, bias, num_groups = dtype_x_and_axis
+    helpers.test_function(
+        ground_truth_backend=ground_truth_backend,
+        fw=backend_fw,
+        test_flags=test_flags,
+        fn_name=fn_name,
+        on_device=on_device,
+        input_dtypes=dtype,
+        x=x[0],
+        num_groups=num_groups,
+        weight=weight,
+        bias=bias,
+        eps=eps,
+        rtol_=1e-03,
+    )
+
+
+# hard_tanh
+@handle_test(
+    fn_tree="functional.ivy.experimental.hard_tanh",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```

Comment:
Please make sure to add the necessary `with_supported_dtypes` decorators to all functions so that we can test for `get_dtypes("valid")`.